ps_viewedproduct.php

```diff
@@ -166,10 +181,11 @@ public function getCacheId($name = null)
         return parent::getCacheId('ps_viewedproduct|' . $key);
     }
 
-    public function renderWidget($hookName = null, array $configuration = [])
+    public function renderWidget($hookName = null, array $configuration = array())
     {
-        if ('displayProductButtons' === $hookName) {
-            $this->addViewedProduct($configuration['product']['id_product']);
+        $this->currentProductId = $configuration['product']['id_product'];
```

Comment:
This information may not be available anywhere, that's why it was inside the `if()` statement